compose/cli/main.py

```diff
@@ -165,15 +165,21 @@ def port(self, project, options):
         Usage: port [options] SERVICE PRIVATE_PORT
 
         Options:
-            --protocol=proto  tcp or udp (defaults to tcp)
+            --protocol=proto  tcp or udp [default: tcp]
             --index=index     index of the container if there are multiple
-                              instances of a service (defaults to 1)
+                              instances of a service [default: 1]
         """
-        service = project.get_service(options['SERVICE'])
+        index = int(options.get('--index')) - 1
+        containers = sorted(
+            project.containers(service_names=[options['SERVICE']], stopped=True) +
+            project.containers(service_names=[options['SERVICE']], one_off=True),
+            key=attrgetter('name'))
         try:
-            container = service.get_container(number=options.get('--index') or 1)
+            container = containers[index]
```

Comment:
I don't think this is right.  This should be handled by `service.get_container()` like the old implementation.